### PR TITLE
[Backport] Added option to exclude discount amount from minimum order amount calculation

### PR DESCRIPTION
**Pull request by ccasciotti**
Original pull request: https://github.com/magento/magento2/pull/19136

### Description

This PR adds option for store manager to include or exclude discount amount from minimum order amount calculation. Sometimes client wants to exclude discount amount from minimum order and want that only subtotal is used. The default option is set to "Yes" so Magento will use the current behavior.

### Fixed Issues (if relevant)

none

### Manual testing scenarios

1. Access backend
2. Go to Stores > Configuration > Sales > Sales > Minimum Order Amount
3. Leave "Yes" in "Include Discount Amount" field to test normal behavior.
4. Set "No" in "Include Discount Amount" field to test new behavior without discount.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=18524) <br/>All committers have signed the CLA.

**Comment by ccasciotti** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @ccasciotti. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @ccasciotti, here is your [new Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/pr-18524/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/pr-18524//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/pr-18524//admin)
Login: `admin` Password: `123123q`

**Comment by ccasciotti** [comment]
@magento-engcom-team manual test was successful, please review code

**Review by orlangur — CHANGES_REQUESTED** [review]
Nowadays all changes must be applied to `2.3-develop` first (see https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules). Please prepare a new pull request, this one should be `on hold` until the latter is merged.

**Comment by ccasciotti** [comment]
Hi @orlangur,
I've ported the change request's commit of the merged #19136 PR here, you can review code and eventually unlock this PR for the `2.2-develop`.

Thanks.

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-3986](https://jira.corp.magento.com/browse/ENGCOM-3986) has been created to process this Pull Request

**Pull request merged**
